libs/wingsdk/src/ex/dynamodb-table.ts

```diff
@@ -0,0 +1,405 @@
+
+/**
+ * Properties for transact write item's delete operation.
+ */
+export interface DynamodbTransactWriteItemDeleteProps {
+  /**
+   * The item to delete.
+   */
+  readonly key: Json;
+}
+
+/**
+ * Dynamodb transact write operation
+ */
+export interface DynamodbTransactWriteItem {
+  /**
+   * A request to perform a put operation.
+   */
+  readonly put?: DynamodbTransactWriteItemPutProps;
+  /**
+   * A request to perform a update operation.
+   */
+  readonly update?: DynamodbTransactWriteItemUpdateProps;
+  /**
+   * A request to perform a delete operation.
+   */
+  readonly delete?: DynamodbTransactWriteItemDeleteProps;
+}
+
+/**
+ * Properties for `DynamodbTable.transactWriteItems`.
+ */
+export interface DynamodbTransactWriteItemsProps {
+  /**
+   * The write transact items.
+   */
+  readonly transactItems: DynamodbTransactWriteItem[];
+}
+
+/**
+ * Inflight interface for `DynamodbTable`.
+ */
+export interface IDynamodbTableClient {
+  /**
+   * Put an item into the table.
+   * @param item data to be inserted.
+   * @param props dynamodb PutItem props.
+   * @inflight
+   */
+  putItem(item: Json, props?: DynamodbTablePutItemProps): Promise<void>;
+
+  /**
+   * Get an item from the table.
+   * @param key key of the item.
+   * @param props dynamodb UpdateItem props.
+   * @inflight
+   */
+  updateItem(key: Json, props?: DynamodbTableUpdateItemProps): Promise<Json>;
+
+  /**
+   * Delete an item from the table.
+   * @param key key of the item.
+   * @inflight
+   */
+  deleteItem(key: Json): Promise<void>;
+
+  /**
+   * Get an item from the table.
+   * @param key key of the item.
+   * @inflight
+   */
+  getItem(key: Json): Promise<Json>;
+
+  /**
+   * Get the table.
+   * @inflight
+   */
+  scan(): Promise<Array<Json>>;
+
+  /**
+   * Perform a synchronous write operation that groups up to 100 action requests.
+   * @param props properties for the transact write items operation.
+   * @inflight
+   */
+  transactWriteItems(props: DynamodbTransactWriteItemsProps): Promise<void>;
+}
+
+/**
+ * List of inflight operations available for `DynamodbTable`.
+ * @internal
+ */
+export enum DynamodbTableInflightMethods {
+  /** `DynamodbTable.putItem` */
+  PUT_ITEM = "putItem",
+  /** `DynamodbTable.updateItem` */
+  UPDATE_ITEM = "updateItem",
+  /** `DynamodbTable.deleteItem` */
+  DELETE_ITEM = "deleteItem",
+  /** `DynamodbTable.getItem` */
+  GET_ITEM = "getItem",
+  /** `DynamodbTable.scan` */
+  SCAN = "scan",
+  /** `DynamodbTable.transactWriteItems` */
+  TRANSACT_WRITE_ITEMS = "transactWriteItems",
+}
+
+/**
+ * Base class for `DynamodbTable` Client.
+ */
+export abstract class DynamodbTableClientBase implements IDynamodbTableClient {
+  /**
+   * @param tableName the table name.
+   * @param attributeDefinitions the table attribute definitions.
+   * @param keySchema the table key schema.
+   */
+  constructor(
+    protected tableName: string,
+    protected attributeDefinitions: Json,
+    protected keySchema: Json
```

Comment:
Dynamodb noob question - why does this need to be a `Json` type? is it possible to pass more than two keys? If there's only a hash key and range key, we could simplify it two parameters:
```suggestion
    protected hashKey: string,
    protected rangeKey: string | undefined,
```
This will also reduce the amount of validation we have to do etc.